lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb

```diff
@@ -190,7 +190,25 @@ def columns
       # rubocop:enable Metrics/PerceivedComplexity
 
       def line_item_includes
-        [{ variant: :product, order: [:bill_address, :shipments] }]
+        [{ variant: :product, order: [:bill_address, :shipments, :order_cycle, :adjustments] }]
+      end
+
+      private
+
+      def admin_and_handling_total(order)
+        order.adjustments.select do |a|
+          a.eligible && a.originator_type == 'EnterpriseFee' && a.source_type != 'Spree::LineItem'
+        end.map(&:amount).sum.to_f
+      end
+
+      def ship_total(order)
+        order.adjustments.select{ |a| a.originator_type == 'Spree::ShippingMethod' }.
+          map(&:amount).sum.to_f
+      end
+
+      def payment_fee(order)
+        order.adjustments.select{ |a| a.originator_type == 'Spree::PaymentMethod' }.
+          map(&:amount).sum.to_f
```

Comment:
this is "copy paste change" to something very similar... :-( 
at least, we must move these methods close to the original methods (because they will change at the same time in the future), these 3 methods have to go to order_decorador. I am not sure about the best names but maybe admin_and_handling_total_alternative...

but better than that would be to replace the original methods. I dont see why we cant replace the original methods in order_decorator. do you?

Reply:
Yeah I wasn't sure how to get around this duplication and increased coupling. We can't just replace the original methods, they're closely linked but not compatible.

Reply:
We could add these as alternate methods in `line_item_decorator` and `order_decorator`. If we try to include this directly in the original methods I think we will have a lot more branching and complexity.

Basically the originals will trigger an additional query to fetch the associated objects (which is generally useful). Here we need the same logic, but applied to eager-loaded associated objects that have already been initialized.

Reply:
Is there some way to check whether or not the associated object have already been fetched and initialized at the point they are needed? Like:

```ruby
if adjustments_already_eager_loaded_and_initialized?
   payment_fee_from_loaded_objects
else
   payment_fee
end
```

Reply:
maybe this? https://stackoverflow.com/questions/1376212/how-to-determine-if-rails-association-is-eager-loaded

Reply:
Okay, I've started adapting the code using `if adjustments.loaded?` and it seems to work well. This would also mean any similar queries that eager-load adjustments should work without triggering additional queries...